data-oriented-programming/declarative-data-processing/main.bal

```diff
@@ -0,0 +1,24 @@
+import ballerina/io;
+import ballerina/http;
+
+type Country record {|
+    string country;
+    string continent;
+    int population;
+    int cases;
+    int deaths;
+|};
+
+http:Client covidClient = check new ("http://localhost:9090");
+
+public function main() returns error? {
+    // Perform data transformation using Ballerina's query language
+    json summary = from var {country, continent, population, cases, deaths} in <Country[]>check covidClient->/countries
```

Comment:
@sameerajayasoma @pcnfernando  Why don't we promote having records instead of json type here?

Reply:
The original thought was to do a transformation. 
Is your concern that even without explicit json conversion, it will be automatically handled at APIs?

Reply:
Yeah.. We usually use the records inside the code to do data manipulation and use the json only via wire.

Reply:
As I see it, there can be other scenarios even without the wire; for instance, storing in json files. Let's get insight from @sameerajayasoma as well